src/data_availability/node_state/verifiers.rs

```diff
@@ -219,6 +223,74 @@ pub fn sp1_proof_verifier(
     Ok(vec![hyle_output])
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+struct ReclaimContext {
+    #[serde(rename = "contextAddress")]
+    context_address: String,
+    #[serde(rename = "contextMessage")]
+    context_message: String,
+    #[serde(rename = "extractedParameters")]
+    extracted_parameters: BTreeMap<String, String>,
+    #[serde(rename = "providerHash")]
+    provider_hash: String,
+}
+
+pub fn reclaim_proof_verifier(
+    proof_bin: &[u8],
+    verification_key: &[u8],
+) -> Result<Vec<HyleOutput>, Error> {
+    // 0x0a is \n, we remove a possible trailing \n
+    let verification_key = if verification_key.ends_with(&[0x0a]) {
+        &verification_key[..verification_key.len() - 1]
+    } else {
+        verification_key
+    };
+    let proof_str = String::from_utf8(proof_bin.into()).expect("Invalid UTF-8");
+
+    let proof: reclaim_rust_sdk::Proof =
+        serde_json::from_str(&proof_str).context("couldnt parse reclaim proof")?;
+
+    let context: ReclaimContext = serde_json::from_str(&proof.claim_data.context)
+        .context("could not deserialize context in reclaim proof")?;
+
+    let reclaim_blob = hyle_contract_sdk::Blob {
+        contract_name: "test_contract".into(),
+        data: hyle_contract_sdk::BlobData(
+            bincode::encode_to_vec(&context.extracted_parameters, bincode::config::standard())
+                .context("Couldn't encode reclaim blob")?,
+        ),
+    };
+
+    if verification_key != context.provider_hash.as_bytes() {
+        return Err(Error::msg("Verification key does not match provider hash"));
```

Comment:
```suggestion
        bail!("Verification key does not match provider hash"))
```
histoire de garder la manière qu'on a de le faire :smile: